src/chunk.c

```diff
@@ -2580,15 +2594,34 @@ ts_chunk_drop_chunks(PG_FUNCTION_ARGS)
 				 errmsg("older_than and newer_than timestamps provided to drop_chunks cannot both "
 						"be NULL")));
 
+	if (schema_name == NULL)
+	{
+		/* default schema is public */
+		schema_name = (NameData *) palloc0(NAMEDATALEN);
+		namestrcpy(schema_name, "public");
+	}
 	ht_oids = ts_hypertable_get_all_by_name(schema_name, table_name, CurrentMemoryContext);
 
 	if (table_name != NULL)
 	{
+		ContinuousAgg *ca = NULL;
 		if (ht_oids == NIL)
```

Comment:
Why isn't `if (ht_oids == NIL)` the outer check here? Seems more reasonable to do this check immediately following `ts_hypertable_get_all_by_name` since it returns this list.

Reply:
restricting dropping  chunks from cagg view only if  the view name is explicitly specified.